pkg/controller/common/helpers.go

```diff
@@ -765,6 +770,30 @@ func CalculateConfigFileDiffs(oldIgnConfig, newIgnConfig *ign3types.Config) []st
 	return diffFileSet
 }
 
+// NewIgnFile returns a simple ignition3 file from just path and file contents.
+// It also ensures the compression field is set to the empty string, which is
+// currently required for handling config merging correctly.
+//
+// Note the default Ignition file mode is 0644, owned by root/root.
+func NewIgnFile(path, contents string) ign3types.File {
+	return NewIgnFileBytes(path, []byte(contents))
+}
+
+// NewIgnFileBytes is like NewIgnFile, but accepts binary data
+func NewIgnFileBytes(path string, contents []byte) ign3types.File {
+	return ign3types.File{
+		Node: ign3types.Node{
+			Path: path,
```

Comment:
_Every single_ non-test user is setting `Overwrite: &true`; so I think it would be worth it to add a `enableOverwrite` boolean to the two helper functions.